src/main/kotlin/features/inventory/buttons/InventoryButtons.kt

```diff
@@ -67,10 +69,11 @@ object InventoryButtons {
 
     @Subscribe
     fun onRenderForeground(it: HandledScreenForegroundEvent) {
-        val bounds = it.screen.getRectangle()
+		if (TConfig.onlyInv && it.screen !is InventoryScreen) return
+		val bounds = it.screen.getRectangle()
 
 		var hoveredComponent: InventoryButton? = null
-        for (button in getValidButtons()) {
+		for (button in getValidButtons()) {
```

Comment:
could you consider just editing the `getValidButtons()` return value instead? it is used as a sort of "single source of truth" for what buttons are active, so other codepaths would also automatically be disabled (exclusion zones, for example)